src/types.h

```diff
@@ -387,6 +387,13 @@ static Janet jaylib_wrap_vec2(Vector2 x) {
     return janet_wrap_tuple(janet_tuple_end(tup));
 }
 
+static Vector2 jaylib_unwrap_vec2(const Janet val) {
+  Janet *tup = janet_unwrap_tuple(val);
+  float x = janet_unwrap_number(tup[0]);
```

Comment:
I would rather use `janet_getindex` here. Does direct indexing on tuple even work?

Reply:
ya direct indexing on tuple works, at least from what I stole from the janet source code lol. Here's an example of straight indexing https://github.com/janet-lang/janet/blob/9fa945ad9301dc099563353dd4e81007bdf40741/src/core/peg.c#L1292

using `janet_getindex` would allow passing both tuples and arrays right? Would that cause any confusion in the situation when changing arr values?

```janet
(var arr @[1 2])
(var cam (camera-2d))
(put cam :offset arr)

# would you expect (cam :offset) to reflect this change?
(put arr 0 3) 
```

Reply:
I've switched to using janet indexed here. Had to use `janet_indexed_view` instead of `janet_getindexed`, since the unwrap function was given the raw janet value instead of argv+n